### PR TITLE
fix: update is-dat-complete script so it works on my machine too

### PR DESCRIPTION
**Pull request by gobengo**
Note:
* this is targeting `feat/is-car-dag-complete` to update #2 

Motivation:
* this is what I did to get the script executing and producing more useful errors, which I'll dig into next

### CODE REVIEW

File: is-car-dag-complete.js [hunk 1]
```diff
@@ -107,17 +113,29 @@ export class S3CarReader {
...
+      Bucket: this._bucketName,
+      Key: path,
+    })
+    const response = await this._s3.send(getCarCommand)
+    const responseCarReader = await CarReader.fromIterable(response.Body);
+    const roots = await responseCarReader.getRoots()
+    debugLog('_readCar read car', { roots, blocks: responseCarReader.blocks() });
+    for await (const block of responseCarReader.blocks()) {
+      debugLog('_readCar writing to blockstore', block.cid.toString())
+      blockstore.set(block.cid.toString(), block);
```
Comment (gobengo) [inline is-car-dag-complete.js:136]: It's possible that this needs to read something to find yet more cids that should go in the blockstore than just the ones it gets from `responseCarReader.blocks()` ? (e.g. is `blocks()` walking a graph for me or do I need to do that manually?)

File: is-car-dag-complete.js [hunk 2]
```diff
@@ -129,9 +147,11 @@ export async function walkDag (root, blockstore) {
   let nextCids = [root]
   while (true) {
     const nextCid = nextCids.shift()
+    debugLog('walkDag iter', nextCid.toString())
     if (!nextCid) return
-    const block = blockstore.get(nextCid.toString())
-    if (!block) throw new Error(`missing block: ${nextCid}`)
+    const blockstoreKey = nextCid.toString();
+    const block = blockstore.get(blockstoreKey)
+    if (!block) throw new Error(`missing block: ${blockstoreKey}`)
```
Comment (gobengo) [inline is-car-dag-complete.js:154]: This throws even for a 'good cid' I uploaded through NFTup (iirc)

```

bengo@bengo ~/protocol.ai/ep-speed-test ⚡  DEBUG='elastic-ipfs' node is-car-dag-complete.js bafybeihgosrdjo6wfdj5bqwzlsctdqzasudneew7xjv5xkqmo7hfcxop3a
🚗 raw/bafybeihgosrdjo6wfdj5bqwzlsctdqzasudneew7xjv5xkqmo7hfcxop3a/nft-82943/ciqdipie7ej6uwdpommyoaasjeyrikvqgbz47edru5cphgkg7vlq75a.car
  elastic-ipfs _readCar read car {
  roots: [ CID(bafybeihgosrdjo6wfdj5bqwzlsctdqzasudneew7xjv5xkqmo7hfcxop3a) ],
  blocks: Object [AsyncGenerator] {}
} +0ms
  elastic-ipfs _readCar writing to blockstore bafybeihgosrdjo6wfdj5bqwzlsctdqzasudneew7xjv5xkqmo7hfcxop3a +2ms
  elastic-ipfs _readCar writing to blockstore bafkreiaqg7tac757i6bdnkz2g3shzuy33ppmhp4b2oklhd54djd6knedu4 +0ms
  elastic-ipfs _readCar writing to blockstore bafkreie3hl5sebdgf4sdtlblrn2h6ywtxlt7ivmih6hxsk6mwo4d6dwz4u +0ms
  elastic-ipfs _readCar writing to blockstore bafkreidz5hr4wqvu7rj3uj4e5me4uqupe6usn4755xgqyau7vmne7qr2i4 +0ms
  elastic-ipfs _readCar writing to blockstore bafkreie3lq2r26dewlxmruvoyy2kx2fqh4m6xnkov4vype2v2wqdbekjee +0ms
  elastic-ipfs _readCar writing to blockstore bafkreiej3ac2za57pcbh7jsvmneiyvgiexoxfhsakvelwwfxbsrveg4h3u +0ms
  elastic-ipfs _readCar writing to blockstore bafkreih5aeeyckxfyayjlkwl37ipfbardwvzpuajn4maxucem4m56cyeam +0ms
  elastic-ipfs _readCar writing to blockstore bafkreibppwbdzkizggcqgqqvvnsgckdm5hzd3fxhype44vrgs3m53ukmba +0ms
  elastic-ipfs _readCar writing to blockstore bafkreif7dkr2wpyvqqryv6u7upon4bh3n6damfrt6qsjphfdcxmkfu5fmm +1ms
  elastic-ipfs _readCar writing to blockstore bafkreihcaakuh7ee3kis7udlq5gf3xyhbgy2yjyaljsaxjr3a46vrd6pya +0ms
  elastic-ipfs _readCar writing to blockstore bafkreih3a2axzedjgjbo57bibdmzkxpeeghhfnefmdf3vrfhz6iuldgtia +0ms
  elastic-ipfs _readCar writing to blockstore bafkreicnfoqupr7gjx5oqy6cnsyitn3m2mpjk65hqehokbqepjdxfecsbm +0ms
  elastic-ipfs _readCar writing to blockstore bafkreigulb53otfvkve5upvrcmtu24v3suyephn4uq6cx6dp4fchzx7sxe +0ms
  elastic-ipfs _readCar writing to blockstore bafkreiemhlqpjm4wn2l5vqa756n6fpx5jvspot45mf23nzpmssqpmjb3rq +0ms
  elastic-ipfs _readCar writing to blockstore bafkreicsdn2xeqw6gggehzbp27yd5yfjcr6vrwreobe7nakfg3pujqcmoi +0ms
  elastic-ipfs _readCar writing to blockstore bafkreiapwosafhzzbyw7yq57rhxmclyqbtddliyreislytuxbb6cder4km +1ms
  elastic-ipfs _readCar writing to blockstore bafkreihioeu4opyo6bqni7lrfdftblucrvtc7oqn4vjakspoxzkeio2yxi +0ms
  elastic-ipfs _readCar writing to blockstore bafkreigkevv3xgo72jr4oyw6tlb5zvz5sql7ufc74jjxhkhgucofecglvy +0ms
  elastic-ipfs _readCar writing to blockstore bafkreibivaqlkqtfknrxh3f5mddltp3yw4lhfe3jqickugmhqwyg2b7m2m +1ms
  elastic-ipfs _readCar writing to blockstore bafkreiges4a4wvwpl5qejzrcgjlsmrbagaxt7pzvnsfmdhrqvxh4qptnay +0ms
  elastic-ipfs _readCar writing to blockstore bafkreifwky3j2xyzoae2r2cymd7dv73k7eazqk7wczqiivwp75hhkw54ra +0ms
  elastic-ipfs _readCar writing to blockstore bafkreieqg27ha2qf2c3xra7xgc5mhdgqbjs27jnr25xngeadndde6jd27q +0ms
  elastic-ipfs _readCar writing to blockstore bafkreid4fc7fsfni5oj343wyyse24l3vlwey2boaxcuyqiekylrij32gw4 +0ms
  elastic-ipfs _readCar writing to blockstore bafkreiefdjom3zkfk5gfe63coa7zu2sbpwe7dsq6w4r2j65wtkdmefzoce +0ms
  elastic-ipfs _readCar writing to blockstore bafkreibv4f5diuaxnp4mr622fy5iwqszq3oeuyg27b3zowlqdcwb4srhzi +0ms
  elastic-ipfs _readCar writing to blockstore bafkreibayp2ndeewb2f3a23sidycud7gh4rs37verc3k6fy4magbgln6am +0ms
  elastic-ipfs _readCar writing to blockstore bafkreiewuutytd3426cmwhns3cvpuway425mnwdtq4zj35mw3v75nf6hs4 +0ms
  elastic-ipfs _readCar writing to blockstore bafkreiaicldcmcig3nsorrxoborqe3htdkcye6l6s7jqyqdprppq7i6nhi +0ms
  elastic-ipfs _readCar writing to blockstore bafkreifyd7psm33ns66m6bhcbauh5uffrhga2kuowgyjktzfru3vzi3eem +0ms
  elastic-ipfs _readCar writing to blockstore bafkreihe43ohnr2k6byxj2t6bk3bpfihwzbjdv5gs3yaccviqeb3y44ttq +0ms
  elastic-ipfs _readCar writing to blockstore bafkreiehw5bosvzo74xgmvpp2veec4v3isruelrm35wybxdafptcy2sv5e +0ms
  elastic-ipfs _readCar writing to blockstore bafkreib6kntwcmjkqmzbnuwns37iobhmv26k5xnrg7jtjf4td6lfe5i7ha +0ms
  elastic-ipfs _readCar writing to blockstore bafkreichmugnhprrh7dpju5wlgyxqt6bi3yjaisifh5gcqdpgiq6cnsriy +0ms
  elastic-ipfs _readCar writing to blockstore bafkreia4og6qdmlm2wsexvcr2h2ycd7u6knnn7plq76bzpa4veiqkk7wkm +0ms
  elastic-ipfs _readCar writing to blockstore bafkreih2rptp6w72rhk6bhnwoisieq5ux5j2rk5zdhm7ttpp2v5y3ymmqy +0ms
  elastic-ipfs _readCar writing to blockstore bafkreiclvbygrf7gvohp53yxm2ovzuaomlqwai6lrckaobeoslz2o42lfe +0ms
  elastic-ipfs _readCar writing to blockstore bafkreica6tnmi2m7zmhupmfgrulksfwaa3trl6hoytq4fqltcx3gjws7lu +0ms
  elastic-ipfs _readCar writing to blockstore bafkreickfvgn7fg3lksusrn3x55b7c5yhzkk3bcfcumagrzcecnjtohf6m +0ms
  elastic-ipfs _readCar writing to blockstore bafkreiexqbsr2pvdmjo4v4nkkttl4atafo6klquzwhsvjrzqiniw3x2yt4 +0ms
  elastic-ipfs _readCar writing to blockstore bafkreif6mys6sksqrykddfxobvhwfrxphbdg5oe4ixzn5evb7kztt7wyni +0ms
🧱 40 new blocks
  elastic-ipfs walkDag iter bafybeihgosrdjo6wfdj5bqwzlsctdqzasudneew7xjv5xkqmo7hfcxop3a +1ms
  elastic-ipfs walkDag iter bafkreiaqg7tac757i6bdnkz2g3shzuy33ppmhp4b2oklhd54djd6knedu4 +3ms
  elastic-ipfs walkDag iter bafkreie3hl5sebdgf4sdtlblrn2h6ywtxlt7ivmih6hxsk6mwo4d6dwz4u +0ms
  elastic-ipfs walkDag iter bafkreidz5hr4wqvu7rj3uj4e5me4uqupe6usn4755xgqyau7vmne7qr2i4 +0ms
  elastic-ipfs walkDag iter bafkreie3lq2r26dewlxmruvoyy2kx2fqh4m6xnkov4vype2v2wqdbekjee +0ms
  elastic-ipfs walkDag iter bafkreiej3ac2za57pcbh7jsvmneiyvgiexoxfhsakvelwwfxbsrveg4h3u +0ms
  elastic-ipfs walkDag iter bafkreih5aeeyckxfyayjlkwl37ipfbardwvzpuajn4maxucem4m56cyeam +0ms
  elastic-ipfs walkDag iter bafkreibppwbdzkizggcqgqqvvnsgckdm5hzd3fxhype44vrgs3m53ukmba +0ms
  elastic-ipfs walkDag iter bafkreif7dkr2wpyvqqryv6u7upon4bh3n6damfrt6qsjphfdcxmkfu5fmm +0ms
  elastic-ipfs walkDag iter bafkreihcaakuh7ee3kis7udlq5gf3xyhbgy2yjyaljsaxjr3a46vrd6pya +0ms
  elastic-ipfs walkDag iter bafkreih3a2axzedjgjbo57bibdmzkxpeeghhfnefmdf3vrfhz6iuldgtia +0ms
  elastic-ipfs walkDag iter bafkreicnfoqupr7gjx5oqy6cnsyitn3m2mpjk65hqehokbqepjdxfecsbm +0ms
  elastic-ipfs walkDag iter bafkreigulb53otfvkve5upvrcmtu24v3suyephn4uq6cx6dp4fchzx7sxe +0ms
  elastic-ipfs walkDag iter bafkreiemhlqpjm4wn2l5vqa756n6fpx5jvspot45mf23nzpmssqpmjb3rq +0ms
  elastic-ipfs walkDag iter bafkreicsdn2xeqw6gggehzbp27yd5yfjcr6vrwreobe7nakfg3pujqcmoi +0ms
  elastic-ipfs walkDag iter bafkreiapwosafhzzbyw7yq57rhxmclyqbtddliyreislytuxbb6cder4km +0ms
  elastic-ipfs walkDag iter bafkreihioeu4opyo6bqni7lrfdftblucrvtc7oqn4vjakspoxzkeio2yxi +0ms
  elastic-ipfs walkDag iter bafkreigkevv3xgo72jr4oyw6tlb5zvz5sql7ufc74jjxhkhgucofecglvy +0ms
  elastic-ipfs walkDag iter bafkreibivaqlkqtfknrxh3f5mddltp3yw4lhfe3jqickugmhqwyg2b7m2m +0ms
  elastic-ipfs walkDag iter bafkreiges4a4wvwpl5qejzrcgjlsmrbagaxt7pzvnsfmdhrqvxh4qptnay +0ms
  elastic-ipfs walkDag iter bafkreifwky3j2xyzoae2r2cymd7dv73k7eazqk7wczqiivwp75hhkw54ra +0ms
  elastic-ipfs walkDag iter bafkreieqg27ha2qf2c3xra7xgc5mhdgqbjs27jnr25xngeadndde6jd27q +0ms
  elastic-ipfs walkDag iter bafkreid4fc7fsfni5oj343wyyse24l3vlwey2boaxcuyqiekylrij32gw4 +0ms
  elastic-ipfs walkDag iter bafkreiefdjom3zkfk5gfe63coa7zu2sbpwe7dsq6w4r2j65wtkdmefzoce +0ms
  elastic-ipfs walkDag iter bafkreibv4f5diuaxnp4mr622fy5iwqszq3oeuyg27b3zowlqdcwb4srhzi +0ms
  elastic-ipfs walkDag iter bafkreibayp2ndeewb2f3a23sidycud7gh4rs37verc3k6fy4magbgln6am +0ms
  elastic-ipfs walkDag iter bafkreiewuutytd3426cmwhns3cvpuway425mnwdtq4zj35mw3v75nf6hs4 +0ms
  elastic-ipfs walkDag iter bafkreiaicldcmcig3nsorrxoborqe3htdkcye6l6s7jqyqdprppq7i6nhi +0ms
  elastic-ipfs walkDag iter bafkreifyd7psm33ns66m6bhcbauh5uffrhga2kuowgyjktzfru3vzi3eem +0ms
  elastic-ipfs walkDag iter bafkreihe43ohnr2k6byxj2t6bk3bpfihwzbjdv5gs3yaccviqeb3y44ttq +0ms
  elastic-ipfs walkDag iter bafkreiehw5bosvzo74xgmvpp2veec4v3isruelrm35wybxdafptcy2sv5e +0ms
  elastic-ipfs walkDag iter bafkreib6kntwcmjkqmzbnuwns37iobhmv26k5xnrg7jtjf4td6lfe5i7ha +0ms
  elastic-ipfs walkDag iter bafkreichmugnhprrh7dpju5wlgyxqt6bi3yjaisifh5gcqdpgiq6cnsriy +0ms
  elastic-ipfs walkDag iter bafkreia4og6qdmlm2wsexvcr2h2ycd7u6knnn7plq76bzpa4veiqkk7wkm +0ms
  elastic-ipfs walkDag iter bafkreih2rptp6w72rhk6bhnwoisieq5ux5j2rk5zdhm7ttpp2v5y3ymmqy +0ms
  elastic-ipfs walkDag iter bafkreiclvbygrf7gvohp53yxm2ovzuaomlqwai6lrckaobeoslz2o42lfe +0ms
  elastic-ipfs walkDag iter bafkreica6tnmi2m7zmhupmfgrulksfwaa3trl6hoytq4fqltcx3gjws7lu +0ms
  elastic-ipfs walkDag iter bafkreickfvgn7fg3lksusrn3x55b7c5yhzkk3bcfcumagrzcecnjtohf6m +0ms
  elastic-ipfs walkDag iter bafkreiexqbsr2pvdmjo4v4nkkttl4atafo6klquzwhsvjrzqiniw3x2yt4 +0ms
  elastic-ipfs walkDag iter bafkreif6mys6sksqrykddfxobvhwfrxphbdg5oe4ixzn5evb7kztt7wyni +0ms
  elastic-ipfs walkDag iter bafkreigu7liubfayn47eolqhf5fmofuzhw5nigqleuttwlkfy6rylmthre +0ms
file:///Users/bengo/protocol.ai/ep-speed-test/is-car-dag-complete.js:154
    if (!block) throw new Error(`missing block: ${blockstoreKey}`)
                      ^

Error: missing block: bafkreigu7liubfayn47eolqhf5fmofuzhw5nigqleuttwlkfy6rylmthre
    at walkDag (file:///Users/bengo/protocol.ai/ep-speed-test/is-car-dag-complete.js:154:23)
    at main (file:///Users/bengo/protocol.ai/ep-speed-test/is-car-dag-complete.js:34:9)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Comment (gobengo) [reply is-car-dag-complete.js:154]: Where I'm 'stuck' now (have to run before digging further) is why this block would be missing for a 'known good upload'.

Comment (alanshaw) [reply is-car-dag-complete.js:154]: `MaxKeys: 1` - https://github.com/alanshaw/ep-speed-test/pull/2/commits/3622c60c6f765830f091abee38ca5a2b21de5f04#diff-135987c416b52da905d9a73a49ae042ca966bfda3454c78e9b59474c9a829849L113

I copied that from the existence check (where only 1 needs to exist).

Comment (alanshaw) [reply is-car-dag-complete.js:154]:
```console
$ node is-car-dag-complete.js bafybeihgosrdjo6wfdj5bqwzlsctdqzasudneew7xjv5xkqmo7hfcxop3a 
🚗 raw/bafybeihgosrdjo6wfdj5bqwzlsctdqzasudneew7xjv5xkqmo7hfcxop3a/nft-82943/ciqdipie7ej6uwdpommyoaasjeyrikvqgbz47edru5cphgkg7vlq75a.car
🧱 40 new blocks
🚗 raw/bafybeihgosrdjo6wfdj5bqwzlsctdqzasudneew7xjv5xkqmo7hfcxop3a/nft-82943/ciqifb7yqf4jsoi5qv7p6763hmezvwieewizku6e2kwd7ptlv5g43kq.car
🧱 17 new blocks
✅ bafybeihgosrdjo6wfdj5bqwzlsctdqzasudneew7xjv5xkqmo7hfcxop3a is a complete DAG on S3
```

**Pull request merged**